src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -269,6 +272,51 @@ def validate_managed_registry(cli_ctx, registry_name, resource_group_name=None,
     return registry, resource_group_name
 
 
+def get_validate_platform(os_type, platform):
+    """Gets and validates the Platform from both flags
+    :param str os_type: The name of OS passed by user in --os flag
+    :param str platform: The name of Platform passed by user in --platform flag
+    """
+    from azure.mgmt.containerregistry.v2018_09_01.models import OS, Architecture, Variant
+    # Defaults
+    platform_os = OS.linux.value
+    platform_arch = Architecture.amd64.value
+    platform_variant = None
+
+    if platform:
+        platform_split = platform.split('/')
+        platform_os = platform_split[0]
+        platform_arch = platform_split[1] if len(platform_split) > 1 else Architecture.amd64.value
+        platform_variant = platform_split[2] if len(platform_split) > 2 else None
+
+    if os_type and platform:
+        if os_type.lower() != platform_os.lower():
+            raise CLIError("The OS in '--platform' should exactly match the value provided in '--os'.")
```

Comment:
This was what I was looking for in the example. If platform contains the OS then why the redundancy?  It would seem you could just accept, `--os` and `--platform`, each with reflected enums, and then concatenate them together if that's the format the service needs. Otherwise, how is someone supposed to know what values of platform they can use?

Reply:
I think that's because we wanted to be consistent with docker's own `--platform` flag (which is currently experimental) https://docs.docker.com/engine/reference/commandline/build/#options
Docker does not have any `os` flag though. I am not sure if the plan is to deprecate `--os` from az acr. //cc: @northtyphoon

Reply:
Yes. We try to provide similar user experience as `docker build` command in `az acr build` command. That's why we introduce a new `--platform` build parameter and it's very specific to build scenarios. But we still need to keep `--os` for backward compatible. It's possible to deprecate `--os` in the future. I think the code paves the way for the future transition.

Reply:
In that case, you can deprecate `--os` now. Just mark it as hidden. It will continue to work but point people toward the new parameter. Also, it should help prevent scenarios where the user supplies both.